processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -181,9 +181,11 @@ public SegmentAnalysis apply(SegmentAnalysis arg1, SegmentAnalysis arg2)
       public byte[] computeCacheKey(SegmentMetadataQuery query)
       {
         byte[] includerBytes = query.getToInclude().getCacheKey();
-        return ByteBuffer.allocate(1 + includerBytes.length)
+        byte[] analysisTypesBytes = query.getAnalysisTypesCacheKey();
```

Comment:
as mentioned elsewhere `0xFF` is generally what we're using as byte array filler between UTF8 stuff.